### PR TITLE
TiFi Plugin Development

### PR DESCRIPTION
**Pull request by OscarYin**

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -31,7 +31,7 @@ jobs:
         with:
           repository: LedgerHQ/ethereum-plugin-sdk
           path: ethereum-plugin-sdk
-          ref: ${{ github.base_ref || github.ref_name }}
+          ref: develop
```
Comment (samuel-tifi) [inline .github/workflows/ci-workflow.yml:34]: May I ask why we need change the `ref` here?

File: src/handle_query_contract_id.c [hunk 2]
```diff
@@ -1,21 +1,69 @@
...
+        case ADD_LIQUIDITY_ETH:
+            strlcpy(msg->version, "Add Liquidity", msg->versionLength);
+            msg->result = ETH_PLUGIN_RESULT_OK;
+            break;
+        case REMOVE_LIQUIDITY_ETH:
+            strlcpy(msg->version, "Remove Liquidity", msg->versionLength);
+            msg->result = ETH_PLUGIN_RESULT_OK;
+            break;
+        case LUCKY_BAG_AND_STAKE_DEPOSIT:
+            strlcpy(msg->version, "DEPOSIT", msg->versionLength);
```
Comment (samuel-tifi) [inline src/handle_query_contract_id.c]: `DEPOSIT` -> `Deposit`
Could we distinguish the deposit is coming from Lucky Bags or Stake?

File: src/handle_query_contract_id.c [hunk 3]
```diff
@@ -1,21 +1,69 @@
...
+            msg->result = ETH_PLUGIN_RESULT_OK;
+            break;
+        case STAKE_WITHDRAW:
+            strlcpy(msg->version, "Unstake", msg->versionLength);
+            msg->result = ETH_PLUGIN_RESULT_OK;
+            break;
+        case LUCKY_BAG_WITHDRAW:
+            strlcpy(msg->version, "Lucky Bag", msg->versionLength);
```
Comment (samuel-tifi) [inline src/handle_query_contract_id.c]: `Luck Bag` -> `Luck Bags`

File: src/handle_query_contract_id.c [hunk 4]
```diff
@@ -1,21 +1,69 @@
...
+            msg->result = ETH_PLUGIN_RESULT_OK;
+            break;
+        case LUCKY_BAG_ENTER:
+            strlcpy(msg->version, "Lucky Bag", msg->versionLength);
```
Comment (samuel-tifi) [inline src/handle_query_contract_id.c]: `Lucky Bag` -> `Lucky Bags`

File: src/handle_query_contract_ui.c [hunk 5]
```diff
@@ -1,88 +1,203 @@
...
+            strlcpy(msg->title, "Withdraw", msg->titleLength);
+            amountToString(context->amount_received,
+                           sizeof(context->amount_received),
+                           WEI_TO_ETHER,
+                           "TIFI ",
+                           msg->msg,
+                           msg->msgLength);
+            break;
+        case LUCKY_BAG_ENTER:
+            strlcpy(msg->title, "Play with", msg->titleLength);
```
Comment (samuel-tifi) [inline src/handle_query_contract_ui.c]: `Play with` -> `Bet with`

File: src/handle_query_contract_ui.c [hunk 6]
```diff
@@ -1,88 +1,203 @@
...
+            msg->msg[1] = 'x';
+            uint64_t chainid = 0;
+            getEthAddressStringFromBinary(
+                context->token_sent,
+                msg->msg + 2,  // +2 here because we've already prefixed with '0x'.
+                msg->pluginSharedRW->sha3,
+                chainid);
+            break;
+        case LOAN_DEPOSIT:
+            strlcpy(msg->title, "Loan Deposite", msg->titleLength);
```
Comment (samuel-tifi) [inline src/handle_query_contract_ui.c]: `Loan Deposite` -> `Loan Deposit`

**Comment by samuel-tifi** [comment]
Should `with` be `withdraw` in this snapshot? https://github.com/samuel-tifi/app-plugin-boilerplate/blob/a86c494b33e9620221aced8f35290a73b07cf254/tests/snapshots/nanos_tifi_add_liquidity_eth/00005.png

File: src/handle_query_contract_id.c (hunk 2 above)
Comment (OscarYin) [reply src/handle_query_contract_id.c]: No we cannot distinguish them since their function ID are same 0xb6b55f25

Comment (OscarYin) [reply src/handle_query_contract_id.c]: I will change the DEPOSIT -> Deposit

File: .github/workflows/ci-workflow.yml (hunk 1 above)
Comment (OscarYin) [reply .github/workflows/ci-workflow.yml:34]: We don't have to use different SDK on the master and develop branch. To get the TIFI token correctly, we should always use the develop branch which contains the latest commits as Adrien suggested.

**Pull request merged**